### PR TITLE
Reflect DiscoveryResponse.control_plane.identifier as a TextReadout stats

### LINKED ISSUE #11000: Reflect DiscoveryResponse.control_plane as a TextReadout stats
Similarly to #10736 we should add a TextReadout stat that will reflect "The control plane instance that sent the response".

### PR DESCRIPTION
**Pull request by efimki**
Add `ControlPlaneStats::identifier` field with text from `DiscoveryResponse.control_plane.identifier` set by the remote.

Only available with GRPC subscription.

Signed-off-by: Misha Efimov <mef@google.com>

Risk Level: Low
Testing: GrpcSubscriptionTestHarness::deliverConfigUpdate includes identifier.
Docs Changes: docs/root/configuration/overview/mgmt_server.rst

Fixes #11000

### CODE REVIEW

**Comment by efimki** [comment]
@ramaraochavali @jmarantz Could you please take a look whether this make sense? 

It is an alternative to #11353 to store control plane identifier in control plane stats instead of subscription stats.

If this makes sense, then I'll update tests and docs. 
Thanks in advance!

File: source/common/config/grpc_mux_impl.cc [hunk 1]
```diff
@@ -124,9 +124,11 @@ bool GrpcMuxImpl::paused(const std::string& type_url) const {
 }
 
 void GrpcMuxImpl::onDiscoveryResponse(
-    std::unique_ptr<envoy::service::discovery::v3::DiscoveryResponse>&& message) {
+    std::unique_ptr<envoy::service::discovery::v3::DiscoveryResponse>&& message,
+    ControlPlaneStats* control_plane_stats) {
   const std::string& type_url = message->type_url();
   ENVOY_LOG(debug, "Received gRPC message for {} at version {}", type_url, message->version_info());
+  control_plane_stats->identifier_.set(message->control_plane().identifier());
```
Comment (ramaraochavali) [inline source/common/config/grpc_mux_impl.cc:131]: nit: check has_control_plane() before using?

File: source/common/config/grpc_stream.h [hunk 2]
```diff
@@ -128,7 +128,8 @@ class GrpcStream : public Grpc::AsyncStreamCallbacks<ResponseProto>,
   ControlPlaneStats generateControlPlaneStats(Stats::Scope& scope) {
```
Comment (ramaraochavali) [inline source/common/config/grpc_stream.h:128]: Can we remove this method and use the new utility method you have added?

File: source/common/config/grpc_mux_impl.cc (hunk 1 above)
Comment (efimki) [reply source/common/config/grpc_mux_impl.cc]: Done.

File: source/common/config/grpc_stream.h (hunk 2 above)
Comment (efimki) [reply source/common/config/grpc_stream.h]: Done.

**Comment by efimki** [comment]
Thanks for your suggestions!
Please take a look when you have a chance.

File: docs/root/configuration/overview/mgmt_server.rst [hunk 3]
```diff
@@ -29,6 +29,7 @@ Management Server has a statistics tree rooted at *control_plane.* with the foll
    connected_state, Gauge, A boolean (1 for connected and 0 for disconnected) that indicates the current connection state with management server
    rate_limit_enforced, Counter, Total number of times rate limit was enforced for management server requests
    pending_requests, Gauge, Total number of pending requests when the rate limit was enforced
+   identifier, TextReadout, The indentifier of the control plane instance that sent the last discovery response
```
Comment (ramaraochavali) [inline docs/root/configuration/overview/mgmt_server.rst:32]: /s/indentifier/identifier

File: docs/root/version_history/current.rst [hunk 4]
```diff
@@ -46,6 +46,7 @@ New Features
 * access loggers: file access logger config added :ref:`log_format <envoy_v3_api_field_extensions.access_loggers.file.v3.FileAccessLog.log_format>`.
 * aggregate cluster: make route :ref:`retry_priority <envoy_v3_api_field_config.route.v3.RetryPolicy.retry_priority>` predicates work with :ref:`this cluster type <envoy_v3_api_msg_extensions.clusters.aggregate.v3.ClusterConfig>`.
 * compressor: generic :ref:`compressor <config_http_filters_compressor>` filter exposed to users.
+* config: added :ref:`identifier <config_cluster_manager_cds>` stat that reflects control plane indentifier.
```
Comment (ramaraochavali) [inline docs/root/version_history/current.rst:49]: nit: s/indentifier/identifier

File: docs/root/configuration/overview/mgmt_server.rst (hunk 3 above)
Comment (efimki) [reply docs/root/configuration/overview/mgmt_server.rst]: Done

File: docs/root/version_history/current.rst (hunk 4 above)
Comment (efimki) [reply docs/root/version_history/current.rst]: Done

**Comment by mattklein123** [comment]
Sorry please merge master again to fix CI.

/wait

**Pull request merged**